### PR TITLE
NO_RCT2 work

### PR DESCRIPTION
**Pull request by janisozaur**
Another batch rebased and remade not to break the version working with vanilla

### CODE REVIEW

File: src/ride/vehicle.c [hunk 1]
```diff
@@ -348,7 +348,7 @@ void vehicle_update_sound_params(rct_vehicle* vehicle)
 							if (vehicle->x != (sint16)0x8000) {
 								int tile_idx = (((vehicle->y & 0xFFE0) * 256) + (vehicle->x & 0xFFE0)) / 32;
 								rct_map_element* map_element;
-								for (map_element = RCT2_ADDRESS(RCT2_ADDRESS_TILE_MAP_ELEMENT_POINTERS, rct_map_element*)[tile_idx]; map_element->type & MAP_ELEMENT_TYPE_MASK; map_element++);
+								for (map_element = gMapElementTilePointers[tile_idx]; map_element->type & MAP_ELEMENT_TYPE_MASK; map_element++);
```
Comment (duncanspumpkin) [inline src/ride/vehicle.c]: Change to get map element

Comment (duncanspumpkin) [inline src/ride/vehicle.c]: Actually this is getting the surface element. Could do with refactoring

Comment (IntelOrca) [inline src/ride/vehicle.c]: `map_element_get_surface`

**Comment by janisozaur** [comment]
updated

**Pull request merged**